Source/player.cpp

```diff
@@ -1414,6 +1415,29 @@ bool IsPlayerAdjacentToObject(Player &player, Object &object)
 	return x <= 1 && y <= 1;
 }
 
+void TryDisarm(Player &player, int i)
+{
+	if (&player == MyPlayer)
+		NewCursor(CURSOR_HAND);
+	if (!Objects[i]._oTrapFlag) {
+		return;
+	}
+	int trapdisper = 2 * player._pDexterity - 5 * currlevel;
+	if (GenerateRnd(100) > trapdisper) {
+		return;
+	}
+	for (int j = 0; j < ActiveObjectCount; j++) {
+		Object &trap = Objects[ActiveObjects[j]];
+		if (trap.IsTrap() && dObject[trap._oVar1][trap._oVar2] - 1 == i) {

```

Comment:
Huh, must've missed this one

```suggestion
		if (trap.IsTrap() && ObjectAtPosition({trap._oVar1,trap._oVar2}) == &Objects[i]) {
```

Removes the last use of `i` in this function so it could take an object reference :)

Reply:
Done and done